internal/engine/uiresource/subscriber.go

```diff
@@ -62,13 +72,38 @@ func (s *Subscriber) OnChange(ctx context.Context, st store.RStore, summary stor
 			// If there's a current version but no last version of this resource,
 			// create it.
 			st.Dispatch(NewUIResourceCreateAction(current))
+			err := s.client.Create(ctx, current.DeepCopy())
+			if err != nil {
+				st.Dispatch(store.NewErrorAction(fmt.Errorf("creating resource %s: %v", current.Name, err)))
+				return
+			}
 			s.lastResources[name] = current
 			continue
 		}
 
 		if !equality.Semantic.DeepEqual(last.Status, current.Status) {
 			// If the current version is different than the last version, update it.
 			st.Dispatch(NewUIResourceUpdateStatusAction(current))
+
+			freshResource := v1alpha1.UIResource{}
+			err := s.client.Get(ctx, types.NamespacedName{Name: current.Name}, &freshResource)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					delete(s.lastResources, name)
+					continue
+				}
+
+				st.Dispatch(store.NewErrorAction(fmt.Errorf("fetching resource %s: %v", current.Name, err)))
+				return
+			}
+
+			freshResource.Status = *freshResource.Status.DeepCopy()
+
+			err = s.client.Status().Update(ctx, &freshResource)
```

Comment:
Are these subscribers the exclusive updaters of resource/session respectively? If not, I think on the update at least it's reasonable to ignore a conflict (and not update internal state).